packages/asset-swapper/src/utils/market_operation_utils/orders.ts

```diff
@@ -271,14 +284,44 @@ export function createBridgeOrder(
         takerToken,
         makerAmount,
         takerAmount,
-        fillData: fill.fillData,
+        fillData: createFinalBridgeOrderFillDataFromCollapsedFill(fill),
         source: fill.source,
         sourcePathId: fill.sourcePathId,
         type: FillQuoteTransformerOrderType.Bridge,
         fills: [fill],
     };
 }
 
+function createFinalBridgeOrderFillDataFromCollapsedFill(fill: CollapsedFill): FillData {
+    switch (fill.source) {
+        case ERC20BridgeSource.UniswapV3: {
+            const fd = fill.fillData as UniswapV3FillData;
+            return {
+                router: fd.router,
+                tokenAddressPath: fd.tokenAddressPath,
+                uniswapPath: getBestUniswapV3PathForInputAmount(fd, fill.input),
+            };
+        }
```

Comment:
The intermediate filldata for univ3 paths (potentially) have a distinct uniswap path (token, fee, token, ...) per sample. The final fillData only needs one, which is the path for the greates sample amount included in the path.

Reply:
What's the advantage of doing this late in the game versus when we decode the sampler output?

I guess we might only want to fill 10% of the requested fill amount on `UniV3` and therefor pruning at sampler output time could mean we pruned the best one for only 10%?

Reply:
yeah, if we only fill 10% then we want the best path for 10%, not 100%